bin/fuel-core/src/cli/run.rs

```diff
@@ -332,20 +344,46 @@ impl Command {
 }
 
 pub async fn exec(command: Command) -> anyhow::Result<()> {
-    let config = command.get_config()?;
     let network_name = {
         #[cfg(feature = "p2p")]
         {
-            config
-                .p2p
-                .as_ref()
-                .map(|config| config.network_name.clone())
+            command
+                .p2p_args
+                .network
+                .clone()
                 .unwrap_or_else(|| "default_network".to_string())
         }
         #[cfg(not(feature = "p2p"))]
         "default_network"
     }
     .to_string();
+
+    #[cfg(feature = "profiling")]
+    let profiling = command.profiling.clone();
+    let config = command.get_config()?;
+
+    // start profiling agent if url is configured
+    #[cfg(feature = "profiling")]
+    let _profiling_agent = profiling
+        .pyroscope_url
+        .as_ref()
+        .map(|url| -> anyhow::Result<_> {
+            // Configure profiling backend
+            let agent = PyroscopeAgent::builder(url, &"fuel-core".to_string())
```

Comment:
Do we want to run it here, or maybe put it along with other services? I'm okay to keep it here, but maybe it is better to follow the same pattern

Reply:
It may make more sense to keep it at the bin level. This agent is more similar to setting up a tracing subscriber, since it's process-wide and there may be unexpected effects caused if multiple nodes have the profiling agent enabled within the same process (i.e. tests).